src/github.com/getlantern/flashlight/genconfig/README.md

```diff
@@ -0,0 +1,83 @@
+# genconfig
+
+**Generate the YAML file and embedded Go datastructures containing the network configuration for the Lantern client**
+
+
+## How it works
+
+Config generation currently follows this approach:
+
+1. Build a *model*, a complete list of working chained servers, masquerades and proxied sites.
+
+1. Fill the Go code templates (*.go), based on the model.
+
+1. Fill the YAML template, based on the model. The YAML template already has filled in the LFS round-robin (flashlight servers for domain fronting).
+
+## Using genconfig
+
+### Setup
+
+You need the s3cmd tool installed and set up.  To install on Ubuntu:
+
+```bash
+sudo apt-get install s3cmd
+```
+
+On OS X:
+
+```bash
+brew install s3cmd
+```
+
+And then run `s3cmd --configure` and follow the on-screen instructions.  You
+can get AWS credentials that are good for uploading to S3 in
+[too-many-secrets/lantern_aws/aws_credential](https://github.com/getlantern/too-many-secrets/blob/master/lantern_aws/aws_credential).
+
+### Managing masquerade hosts
+
+The file domains.txt contains the list of masquerade hosts we use, and
+blacklist.txt contains a list of blacklisted domains that we exclude even if
+present in domains.txt.
+
+To alter the list of domains or blacklist:
+
+1. Edit [`domains.txt`](genconfig/domains.txt) and/or [`blacklist.txt`](genconfig/blacklist.txt)
+2. `go run genconfig.go -domains domains.txt -blacklist blacklist.txt`.
+3. Commit the changed [`masquerades.go`](config/masquerades.go) and [`cloud.yaml`](genconfig/cloud.yaml) to git if you want.
+4. *[Deprecated: we are not doing it any more]* Upload cloud.yaml to s3 using [`udpateyaml.bash`](genconfig/updateyaml.bash) if you want.
```

Comment:
Yeah, the analogous step would be to use `cfg2redis.py` to upload the new configuration if desired.